src/library/scala/collection/GenTraversableOnce.scala

```diff
@@ -96,6 +96,12 @@ trait GenTraversableOnce[+A] extends Any {
    */
   def size: Int
 
+   /** The size of this $coll if it is can be cheaply computed
+   *
+   *  @return    the number of elements in this $coll, or -1 if the size cannot be determined cheaply
+   */
+  protected[collection] def sizeHint: Int = -1
```

Comment:
I think it has done a good job in this use-case: https://github.com/non/psp-view/blob/master/core/src/main/scala/SizeInfo.scala#L8